src/vos/vos_obj_cache.c

```diff
@@ -228,45 +67,15 @@ vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
 		DP_UUID(cont->vc_id), DP_UOID(oid),
 		create ? "true" : "false", epr->epr_lo, epr->epr_hi);
 
-	/* Create the key for obj cache */
-	lkey.olk_cont = cont;
-	lkey.olk_oid = oid;
 
-	rc = daos_lru_ref_hold(occ, &lkey, sizeof(lkey), cont, &lret);
-	if (rc)
-		D_GOTO(failed_2, rc);
-
-	obj = container_of(lret, struct vos_object, obj_llink);
-
-	if (obj->obj_zombie)
-		D_GOTO(failed, rc = -DER_AGAIN);
-
-	if (intent == DAOS_INTENT_KILL) {
-		if (vos_obj_refcount(obj) > 2)
-			D_GOTO(failed, rc = -DER_BUSY);
```

Comment:
Seems these 'refcount' & 'zombie' was used for preventing an object being deleted by vos_obj_delete() while it's still holding by someone else, is there any equivalent in this patch?

Reply:
Yeah, this is probably a little too simplistic.   @mchaarawi gets really good IO500 results with this patch compared to master branch which is why I've cleaned it up a bit to upload.   I wonder if we can leverage the LRU timestamp cache.  I'll poke around a bit.